src/ray/gcs/redis_gcs_client.cc

```diff
@@ -70,98 +70,86 @@ namespace ray {
 
 namespace gcs {
 
-AsyncGcsClient::AsyncGcsClient(const std::string &address, int port,
-                               const ClientID &client_id, CommandType command_type,
-                               bool is_test_client = false,
-                               const std::string &password = "") {
+RedisGcsClient::RedisGcsClient(const ClientOption &option, const ClientInfo &info)
+    : GcsClientInterface(option, info) {
+}
+
+Status RedisGcsClient::Connect(boost::asio::io_service &io_service) {
+  RAY_CHECK(!is_connected_);
+
+  if (option_.server_ip_.empty()) {
+    RAY_LOG(ERROR) << "connect failed, gcs service address is empty.";
+    return Status::Invalid("gcs service address is invalid!");
+  }
+
   primary_context_ = std::make_shared<RedisContext>();
 
   RAY_CHECK_OK(
-      primary_context_->Connect(address, port, /*sharding=*/true, /*password=*/password));
+      primary_context_->Connect(option_.server_ip_, option_.server_port_,
+                                /*sharding=*/true, /*password=*/option_.password_));
 
-  if (!is_test_client) {
+  if (!option_.is_test_client_) {
     // Moving sharding into constructor defaultly means that sharding = true.
     // This design decision may worth a look.
     std::vector<std::string> addresses;
     std::vector<int> ports;
     GetRedisShards(primary_context_->sync_context(), addresses, ports);
-    if (addresses.size() == 0 || ports.size() == 0) {
-      addresses.push_back(address);
-      ports.push_back(port);
+    if (addresses.size() == 0) {
```

Comment:
`addresses.empty()` will be more graceful

Reply:
done